actors/builtin/miner/miner_actor.go

```diff
@@ -1697,6 +1700,12 @@ func handleProvingDeadline(rt Runtime) {
 			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock penalty")
 			penaltyTotal = big.Add(penaltyTotal, penalty)
 
+			// Save modified deadline state.
+			if detectedAny {
+				deadline.Partitions, err = partitions.Root()
+				builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to store partitions")
+			}
```

Comment:
Previously, the `PopExpiredSectors` below re-loaded the partitions and missed these changes, resulting in duplicate power loss.